storage/shard.go

```diff
@@ -1075,6 +1075,7 @@ func (s *dbShard) FetchBlocks(
 	ctx context.Context,
 	id ident.ID,
 	starts []time.Time,
+	blockDuration time.Duration,
```

Comment:
As per previous comment, instead of passing the block duration / block size here,  let's just make the underlying blocks and block retriever return an `xio.BlockReader`.